Mock not working in 12.1.0

<!--
Please make sure you have searched previous issues before opening a new issue.

### If you're having a configuration error

Configuring google sign in can sometimes be tricky. If you're hitting a wall, you can get in touch with a maintainer ([@vonovak](https://github.com/vonovak)) via his [personal site](https://react-native-training.eu/). Please note that this is a paid service.

-->

<!-- Describe your issue in detail. -->
After updating to `12.1.0`, the mock is causing an error when running my tests

`Cannot find module '../src/types' from 'node_modules/@react-native-google-signin/google-signin/jest/build/setup.js'`

Here is what the module looks like when installed

![image](https://github.com/react-native-google-signin/google-signin/assets/30590984/7f29f511-0fba-4402-ad96-ffcd6f7ba382)

inside `jest/build/setup.ts`
![image](https://github.com/react-native-google-signin/google-signin/assets/30590984/ca0f0342-85b7-4af1-a8a5-b814c966bd10)

## Repository with code that reproduces the issue

- You **must** provide a **minimal and easy-to-run** reproduction of your issue - [how to create a minimal reproduction?](https://stackoverflow.com/help/mcve). We're a small team of maintainers and do not have time to try reproduce bugs ourselves.

- Please try to reproduce the bugs on the provided example app. Either provide a link to the repo that reproduces the bug or provide your version of the [`index.js` file](https://github.com/react-native-community/google-signin/blob/master/example/index.js) that reproduces the issue and that we can easily use.

- Explain the steps we need to take to reproduce the issue.

- Include a gif if you think it may help.

- If you're asking for help, please make it easy for us to help you!

## Expected Behavior

<!-- Write what you thought would happen. -->
mock works
## Actual Behavior
mock doesn't work
<!-- Write what happened. Include a gif if needed. If this is a regression, let us know. -->

## Environment

Please provide

- react-native version `0.73.8`
- @react-native-google-signin/google-signin version `12.1.0`
- your Podfile, output of running `pod install` and cocoapods version (if applicable)
- version of Android, gradle, android plugin for gradle, `play-services-auth` and version of the Google Play Services app on your phone (if applicable)
- version of iOS (if applicable)

--- a/jest/setup.tsx
+++ b/jest/setup.tsx
@@ -1,6 +1,6 @@
 import React from 'react';
 import { Pressable, Text } from 'react-native';
-import {
+import type {
   GoogleSigninButton,
   GoogleSigninButtonProps,
   User,

--- a/package.json
+++ b/package.json
@@ -24,7 +24,7 @@
     "app.plugin.js",
     "expo-module.config.json",
     "plugin/build",
-    "jest/build/setup.js",
+    "jest/build",
     "README.md"
   ],
   "scripts": {

